datafusion/src/datasource/file_format/mod.rs

```diff
@@ -32,42 +32,51 @@ use crate::logical_plan::Expr;
 use crate::physical_plan::{Accumulator, ExecutionPlan, Statistics};
 
 use async_trait::async_trait;
-use futures::Stream;
-
-/// A stream of String that can be used accross await calls
-pub type StringStream = Pin<Box<dyn Stream<Item = String> + Send + Sync>>;
-
-/// Convert a vector into a stream
-pub fn string_stream(strings: Vec<String>) -> StringStream {
-    Box::pin(futures::stream::iter(strings))
+use futures::{Stream, StreamExt};
+
+use super::object_store::{FileMeta, ObjectReader, ObjectReaderStream, ObjectStore};
+
+/// The configurations to be passed when creating a physical plan for
```

Comment:
This is a neat idea. 

Is the idea that this information will be provided by the DataSource (e.g. the `ListingTable`)? 

If so I think that makes sense as things like the `statistics` and `schema` may be known via some other cache / catalog mechanism